jep/0000/README.adoc

```diff
@@ -0,0 +1,358 @@
+
+==== LTS Backporting
+
+All backporting will be done according to the link:https://jenkins.io/download/lts/#backporting-process[LTS Backporting Process].
+
+There is no plan to backport changes for the end of Java 8 support to previous LTS baselines.
+
+== Motivation
+
+Java 11 was released on September 25, 2018.
+It is an LTS version with a long support timeline.
+Java 8 was released in March 2014.
+The link:https://www.oracle.com/java/technologies/java-se-support-roadmap.html[Oracle Java SE Support Roadmap] states that premier support for Java 8 ends in March 2022.
+Java 8 enhancements have stopped as effort focuses on Java 11, Java 17, and beyond.
+
+Removing support for Java 8 simplifies the supported configurations and allows further modernization of Jenkins core.
+Ending support for Java 8 allows Jenkins core and Jenkins plugins to use libraries that support Java 11 but do not support Java 8.
+
+== Reasoning
+
+“Goals and non-goals” section in the specification lists design decisions taken
+to ensure it can be delivered by a small team.
+Non-goals in the specification are defined to limit the scope of work.
+The main objective is to move Jenkins core development to Java 11.
+There will be follow-up tasks for further improvements and to adopt new features.
+
+=== Support of Java 17
+
+This JEP intentionally limits its scope by not including Java 17 support.
+It does not prevent work on Java 17, but that work is outside the scope of this JEP.
+
+=== Docker image labeling
+
+Docker image labels were updated in August 2021 to use link:https://www.jenkins.io/blog/2021/08/17/docker-images-use-jdk-11-by-default/[Java 11 by default].
+The image labels that do not explicitly mention a Java version (like `latest`, `lts`, `slim`, `alpine`) are already delivering Java 11.
+
+Image labels that explicitly mention `jdk8` will not be updated after Jenkins core ends support for Java 8.
+
+== Backwards Compatibility
+
+The following backward compatibility requirements are defined:
+
+* Jenkins core and updated plugins should fully support Java 11
+* Jenkins plugins may continue to compile with Java 8 so long as the plugins run successfully with Java 11
+* Jenkins plugins that require a Jenkins version that does not support Java 8 will be expected to compile with Java 11
+
+== Security
+
+=== Process
+
+Only Java 11 with the latest security fixes will be supported at the moment of the public release.
+
+Jenkins security issues on the release that ends support of Java 8 will be processed according to the
+standard link:https://jenkins.io/security/[Jenkins Security Process].
+
+=== Security risks
+
+* No additional security risks are expected due to Jenkins ending support for Java 8
+
+== Infrastructure Requirements
+
+=== ci.jenkins.io
+
+* Tool Infrastructure should continue to offer the latest version of Java 11
+
+=== Jenkins Pipeline Library
+
+* `buildPlugin()`, `runATH()`, and `runPCT()` will run tests with JDK 11
```

Comment:
And `buildPlugin` will compile _using_ JDK 11, even if the plugin is still compatible with 8. `-release` should ensure that the plugin does not accidentally use post-8 APIs. It might however begin implicitly relying on behavioral changes between 8 and 11, resulting in a functional incompatibility with older core versions running on 8.